README.md

```diff
@@ -70,28 +65,48 @@ nix run .#start-vm
 This starts a local VM running in QEMU. Handy to check everything works as
 expected before deploying.
 
-It will forward port 443 of the VM onto port 8443, meaning you may visit
-https://localhost:8443/ once the VM has started.
+When starting, this command will obtain a Vault token and inject it into the VM
+to be used for fetch GitHub client ID and secrets. If needed, you may be
+prompted for your GitHub personal access token.
 
-Nginx is configured using a self-signed certificate, which will cause some
-browser warnings.
+Port 443 of the VM is exposed as port 8443, meaning you may visit
+https://localhost:8443/ once the VM has started. nginx is configured using a
+self-signed certificate, which will cause some browser warnings.
 
-Packit is configured to use basic authentication, but no users exist by default.
-In the VM console, run the following:
+Packit is configured to use GitHub authentication. If needed, after logging in,
+you may grant yourself more permissions on a particular Packit instance through
+the VM console.
 
 ```
-create-basic-user <instance> "admin@localhost.com" password
-grant-role <instance> "admin@localhost.com" ADMIN
+grant-role <instance> <github username> ADMIN
 ```
 
 ## How do I run the integration tests?
 
 ```sh
-nix flake check
+nix flake check -L
 ```
 
-This doesn't test that much yet, just that the Packit API eventually comes up.
-We should at least try to interact with the API a little.
+Depending on your host system and how Nix was installed on it, this may fail
+with a "qemu-kvm: failed to initialize kvm: Permission denied" error. This
+typically means that `/dev/kvm` and is not writable by the Nix build users.
+
+This can be fixed by changing the devices ACLs to make it writable by all
+members of the nixbld group:
+
+```sh
+sudo setfacl -m g:nixbld:rw /dev/kvm
+```
+
+The above command will probably not persist across reboots. For that to work,
+create a udev rule using the following commands:
+
+```sh
+sudo tee /etc/udev/rules.d/50-nixbld-kvm.rules <<EOF
+KERNEL=="kvm", RUN+="/bin/setfacl -m g:nixbld:rw $env{DEVNAME}"
+EOF
```

Comment:
sorry for the questions, just trying to work out how this is equivalent, the `/bin/setfacl -m g:nixbld:rw` is fine and then we want `/dev/kvm` the `$env{DEVNAME}` gives the `/dev` bit but then how does the `/kvm` come into play, youve done `KERNEL=="kvm"` but couldnt find anything related to kernel on `setfacl` man page

Reply:
This is a udev rule, which is how modern (last 10-15 years) Linux distros manage devices in `/dev`. `/dev` is now a temporary in-memory filesystem, and any changes to it don't persist reboot.

The rule says, for a device that has the label `KERNEL=="kvm"`, run the following command. `DEVNAME` is set to `/dev/kvm`. This is more a udev thing than it is a `setfacl` one.